ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/RedundantLambdaAnalyzer.fs

```diff
@@ -0,0 +1,44 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.CommonErrors
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.Util
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+
+[<ElementProblemAnalyzer(typeof<ILambdaExpr>, HighlightingTypes = [| typeof<RedundantParenExprWarning> |])>]
+type RedundantLambdaAnalyzer() =
+    inherit ElementProblemAnalyzer<ILambdaExpr>()
+        
+    let rec compareArgs (lambdaArgs: IFSharpPattern seq) (bodyArgs: IFSharpExpression seq) =
+        if bodyArgs.IsEmpty() then lambdaArgs.IsEmpty() else
+        let equal =
+            match (bodyArgs |> Seq.head).IgnoreInnerParens() with
+            | :? ITupleExpr as tuple ->
+                match (lambdaArgs |> Seq.head) with
+                | :? ITuplePat as pat -> compareArgs pat.Patterns tuple.Expressions
+                | :? ILocalAsPat as pat -> compareArgs [pat.Pattern] [tuple]
+                | :? IParenPat as pat -> compareArgs [pat.Pattern] [tuple]
+                | _ -> false
+            | :? IReferenceExpr as reference ->
+                match lambdaArgs |> Seq.head with
+                | :? ILocalReferencePat as pat -> pat.CompiledName = reference.ShortName
```

Comment:
Please use `ShortName`/`DeclaredName`/`SourceName` instead of `CompiledName`. We're interested in the name in the source code, not how it's compiled.